src/language/records.md

```diff
@@ -0,0 +1,160 @@
+so fields do not have setters. 
+
+Named fields expose getters of the same name. Positional fields expose getters
+of the name `$<position>`, skipping named fields:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');
+
+print(record.$1);    // Prints '123'
+print(record.a);     // Prints 2
+print(record.b);     // Prints true
+print(record.$2);    // Prints 'last'
+```
+
+To streamline record field access even more, see the page on [Patterns][].
+
+{% comment %}
+    TODO: link to patterns page, specifically records destructuring section.
+{% endcomment %}
+
+## Record types
+
+There is no type declaration for individual record types. Records are structurally
+typed based on the types of their fields. A record's _shape_ (the set of its fields,
+the fields' types, and their names, if any) uniquely determines the type of a record. 
+
+Each field in a record has its own type, which can be different from other
+fields' types in the same record. The type system is aware of each
+field's type wherever it is accessed from the record:
+
+```dart
+(num, Object) pair = (42, "a");
+
+var first = pair.$1;     // static type `num`, runtime type `int`
+var second = pair.$2;    // static type `Object`, runtime type `String`
+```
+
+If two unrelated libraries create records with the same set of fields,
+the type system understands that those records are the same type even though the
+libraries are not coupled to each other.
+
+## Record equality
+
+Two records are equal if they have the same _shape_ (set of fields),
+and their corresponding fields have the same values.
+Since named field _order_ is not part of a record's shape, the order of named
+fields does not affect equality:
+
+```dart
+(int x, int y, int z) point = (1, 2, 3);
+(int r, int g, int b) color = (1, 2, 3);
+
+// OK:
+point = color;
+```
+
+```dart
+({int x, int y, int z}) point = (x: 1, y: 2, z: 3);
+({int r, int g, int b}) color = (r: 1, g: 2, b: 3);
+
+// Error, different types:
+point = color;
+```
+
+Records automatically define `hashCode` and `==` methods structurally based on
+their fields.
+
+## Multiple returns
+
+Records allow functions to return multiple values bundled together.
+Using [pattern matching][], you can retrieve record values from a return by
+destructuring them directly into local variables:
+
+{% comment %}
+    TODO: link to patterns page, specifically records destructuring section.
+{% endcomment %}
+
+```dart
+// Returns multiple values in a record:
+(String, int) userInfo(Map<String, dynamic> json) {
+  return (json['name'] as String, json['age'] as int);
+}
+
+void main() {
+  // Destructures using a record pattern:
+  var (name, age) = userInfo(json);
+
+  /* Equivalent to:
+  var info = userInfo(json);
+  var name = info.$1;
+  var age  = info.$2;
+  */
+}
+```
+
+Other ways to accomplish multiple returns are
+either to create a class (which is more verbose), or use
+another collection type like `List` or `Map` (which loses type safety).
```

Comment:
**issue:** Avoid parentheses per [GSG](https://developers.google.com/style/parentheses). Try to ease reading per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
```suggestion
You can accomplish multiple returns in other ways that have some downsides.
* Create a class. This is more verbose.
* Use another collection type like `List` or `Map`. This loses type safety.
```

Reply:
I don't want to use the bullet list because I feel like it draws too much attention to something that's just kind of a throwaway comment and not about records. Maybe see if the edit in the new commit is reasonable?